In [1]:
import numpy as np 
import pandas as pd 

In [2]:
import json 
from sklearn import preprocessing

In [3]:
import matplotlib.pyplot as plt

In [4]:
%load_ext ipython_memory_usage

Enabling IPython Memory Usage, use %imu_start to begin, %imu_stop to end


In [5]:
%imu_start 

'IPython Memory Usage started'

In [5] used 0.4 MiB RAM in 0.12s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 153.6 MiB


### EDA de la dataFrame 'Athlete_events'

In [6]:
df_athlete=pd.read_csv('data/athlete_events.csv',sep=',')

df_athlete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282985              ZOU Jingyuan   M  23.0    1.58     NaN   
282986          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282987            ZUEV Alexander   M  25.0    1.93     NaN   
282988          ZVEREV Alexander   M  24.0    1.98     NaN   
282989              ZYKOVA Yulia   F  26.0     NaN     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282985  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282986                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282987                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   
282988                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   
282989                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  
0                Basketball       Basketball Men's Basketball     NaN  
1                      Judo      Judo Men's Extra-Lightweight     NaN  
2                  Football           Football Men's Football     NaN  
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold  
4             Speed Skating  Speed Skating Women's 500 metres     NaN  
...                     ...                               ...     ...  
282985  Artistic Gymnastics               Men's Parallel Bars    Gold  
282986             Football                               Men  Silver  
282987       3x3 Basketball                               Men  Silver  
282988               Tennis                     Men's Singles    Gold  
282989             Shooting       50m Rifle 3 Positions Women  Silver  

[282990 rows x 14 columns]

In [6] used 114.9 MiB RAM in 5.27s (system mean cpu 15%, single max cpu 90%), peaked 0.0 MiB above final usage, current RAM usage now 268.5 MiB


### DataFrame dictionnary

In [7]:
df_dictionnary=pd.read_csv('data/dictionary.csv')

df_dictionnary=df_dictionnary.rename(columns={'GDP per Capita':'PIB/habitant','Code':'NOC'})

In [7] used -70.6 MiB RAM in 0.71s (system mean cpu 15%, single max cpu 100%), peaked 70.6 MiB above final usage, current RAM usage now 197.9 MiB


### On cherche les athlètes qui n'ont pas de team et ceux des îles TUVALU

In [8]:
df_athlete[df_athlete['NOC']=='UNK']


Name Sex  Age  Height  Weight     Team  NOC        Games  \
61080   Fritz Eccard   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   
130721     A. Laffen   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   

        Year  Season       City             Sport  \
61080   1912  Summer  Stockholm  Art Competitions   
130721  1912  Summer  Stockholm  Art Competitions   

                                      Event Medal  
61080   Art Competitions Mixed Architecture   NaN  
130721  Art Competitions Mixed Architecture   NaN

In [8] used 3.3 MiB RAM in 0.15s (system mean cpu 13%, single max cpu 70%), peaked 0.0 MiB above final usage, current RAM usage now 201.2 MiB


Fritez Eccard a participé dans l'équipe Suédoise et A.Laffen .... aucunes traces je sais pas qui est ce mec 

In [9]:
df_athlete[df_athlete['NOC']=='ROT'].nunique()


Name      10
Sex        2
Age        7
Height     7
Weight     9
Team       1
NOC        1
Games      1
Year       1
Season     1
City       1
Sport      3
Event     12
Medal      0
dtype: int64

In [9] used 1.0 MiB RAM in 0.14s (system mean cpu 22%, single max cpu 57%), peaked 0.0 MiB above final usage, current RAM usage now 202.2 MiB


12 participations parmi 10 athlètes qui ont participé aux JO ces dernières années sous la team des athlètes réfugiés. 

In [10]:
df_athlete=df_athlete.query(" NOC !='ROC' and NOC != 'UNK' ")

In [10] used 26.5 MiB RAM in 0.18s (system mean cpu 11%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 228.7 MiB


## Base données région 

In [63]:
df_regions=pd.read_csv('data/noc_regions.csv')

df_regions=df_regions.rename(columns={'region':'Country'})
df_regions

NOC      Country                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

In [63] used -1.2 MiB RAM in 0.13s (system mean cpu 16%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 27.0 MiB


In [64]:
df_regions.nunique()

NOC        230
Country    206
notes       21
dtype: int64

In [64] used 1.5 MiB RAM in 0.12s (system mean cpu 17%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 28.5 MiB


Il y a des régions qui ont le même NOC 

In [65]:
duplicates=[df_regions.duplicated(subset='Country')]

In [65] used 0.3 MiB RAM in 0.11s (system mean cpu 13%, single max cpu 22%), peaked 0.0 MiB above final usage, current RAM usage now 28.8 MiB


In [66]:
df_regions[df_regions.duplicated(subset='Country')]

NOC         Country                   notes
12   AUS       Australia                     NaN
54   CZE  Czech Republic                     NaN
76   GDR         Germany                     NaN
79   GER         Germany                     NaN
81   GRE          Greece                     NaN
88   HKG           China               Hong Kong
126  MAS        Malaysia                     NaN
143  NBO        Malaysia            North Borneo
147  NFL          Canada            Newfoundland
171  RUS          Russia                     NaN
173  SAA         Germany                     NaN
185  SRB          Serbia                     NaN
197  TCH  Czech Republic                     NaN
208  TUV             NaN                  Tuvalu
210  UAR           Syria    United Arab Republic
213  UNK             NaN                 Unknown
214  URS          Russia                     NaN
222  VNM         Vietnam                     NaN
223  WIF        Trinidad  West Indies Federation
225  YEM           Yemen                     NaN
226  YMD           Yemen             South Yemen
227  YUG          Serbia              Yugoslavia
229  ZIM        Zimbabwe                     NaN

In [66] used 0.4 MiB RAM in 0.12s (system mean cpu 13%, single max cpu 23%), peaked 0.0 MiB above final usage, current RAM usage now 29.2 MiB


In [67]:
df_regions.isnull().sum()

NOC          0
Country      3
notes      209
dtype: int64

In [67] used 0.2 MiB RAM in 0.11s (system mean cpu 11%, single max cpu 21%), peaked 0.0 MiB above final usage, current RAM usage now 29.5 MiB


In [68]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [68] used 0.0 MiB RAM in 0.12s (system mean cpu 16%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 29.5 MiB


Il y a  3 des regions qui ne sont pas attribuées, ROT ( refugee olympic team)  , TUVALU ( 9 archipels qui constituent un État ) et UNK qui correspond à Unknown 

### On ajoute les iles TUVALU 

In [69]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [69] used -2.0 MiB RAM in 0.11s (system mean cpu 16%, single max cpu 25%), peaked 0.0 MiB above final usage, current RAM usage now 27.5 MiB


In [70]:
df_regions=df_regions.query(" NOC != 'ROT' and NOC != 'UNK' ")

df_regions.isnull().sum()

NOC          0
Country      1
notes      209
dtype: int64

In [70] used 2.4 MiB RAM in 0.14s (system mean cpu 12%, single max cpu 20%), peaked 0.0 MiB above final usage, current RAM usage now 29.9 MiB


In [71]:
df_regions['Country']=df_regions['Country'].fillna('Tuvalu')

In [71] used -5.0 MiB RAM in 0.12s (system mean cpu 14%, single max cpu 23%), peaked 0.0 MiB above final usage, current RAM usage now 24.9 MiB


<ipython-input-71-0f7f308fe64b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_regions['Country']=df_regions['Country'].fillna('Tuvalu')


In [73]:
df_regions[df_regions['Country']=='Tuvalu']

NOC Country   notes
208  TUV  Tuvalu  Tuvalu

In [73] used 2.1 MiB RAM in 0.13s (system mean cpu 16%, single max cpu 25%), peaked 0.0 MiB above final usage, current RAM usage now 43.1 MiB


In [74]:
df_regions.isnull().sum()

NOC          0
Country      0
notes      209
dtype: int64

In [74] used 0.4 MiB RAM in 0.11s (system mean cpu 17%, single max cpu 26%), peaked 0.0 MiB above final usage, current RAM usage now 43.6 MiB


# On va merge les dataframes 

## Merge en left 

In [75]:
df_complete=pd.merge(pd.merge(df_athlete,df_regions,on='NOC',how='left'),df_dictionnary,on='NOC',how='left')
#df_complete=df_complete.drop(labels='',axis=1)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  \
0                Basketball       Basketball Men's Basketball     NaN   
1                      Judo      Judo Men's Extra-Lightweight     NaN   
2                  Football           Football Men's Football     NaN   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold   
4             Speed Skating  Speed Skating Women's 500 metres     NaN   
...                     ...                               ...     ...   
282622            Taekwondo                       Women -57kg    Gold   
282623  Artistic Gymnastics                        Men's Team  Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars    Gold   
282625             Football                               Men  Silver   
282626               Tennis                     Men's Singles    Gold   

          Country_x notes      Country_y    Population  PIB/habitant  
0             China   NaN          China  1.371220e+09   8027.683810  
1             China   NaN          China  1.371220e+09   8027.683810  
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085  
...             ...   ...            ...           ...           ...  
282622          USA   NaN  United States  3.214188e+08  56115.718426  
282623        China   NaN          China  1.371220e+09   8027.683810  
282624        China   NaN          China  1.371220e+09   8027.683810  
282625        Spain   NaN          Spain  4.641827e+07  25831.582305  
282626      Germany   NaN        Germany  8.141314e+07  41313.313995  

[282627 rows x 19 columns]

In [75] used 130.4 MiB RAM in 0.39s (system mean cpu 20%, single max cpu 50%), peaked 0.0 MiB above final usage, current RAM usage now 174.0 MiB


In [76]:
df_complete=df_complete.query(" NOC !='ROC' ")

In [76] used -46.5 MiB RAM in 0.23s (system mean cpu 13%, single max cpu 27%), peaked 0.0 MiB above final usage, current RAM usage now 127.5 MiB


In [77]:
#df_complete.info()

In [77] used 0.9 MiB RAM in 0.10s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 128.3 MiB


282 627 lignes , c'est normal on a enlevé les  réfugiés

In [79]:
df_complete.loc[(df_complete['Medal'].isna(),'Medal')]='Pas_medaille'



In [79] used -57.8 MiB RAM in 0.14s (system mean cpu 15%, single max cpu 24%), peaked 32.0 MiB above final usage, current RAM usage now 66.9 MiB


## Encoding test 

In [80]:
dictio_score_medaille={'Pas_medaille':0,
                      'Bronze':1,
                      'Silver':2,
                      'Gold':3}

df_complete['Score_medaille']=df_complete['Medal'].map(dictio_score_medaille)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event         Medal  \
0                Basketball       Basketball Men's Basketball  Pas_medaille   
1                      Judo      Judo Men's Extra-Lightweight  Pas_medaille   
2                  Football           Football Men's Football  Pas_medaille   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War          Gold   
4             Speed Skating  Speed Skating Women's 500 metres  Pas_medaille   
...                     ...                               ...           ...   
282622            Taekwondo                       Women -57kg          Gold   
282623  Artistic Gymnastics                        Men's Team        Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars          Gold   
282625             Football                               Men        Silver   
282626               Tennis                     Men's Singles          Gold   

          Country_x notes      Country_y    Population  PIB/habitant  \
0             China   NaN          China  1.371220e+09   8027.683810   
1             China   NaN          China  1.371220e+09   8027.683810   
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085   
...             ...   ...            ...           ...           ...   
282622          USA   NaN  United States  3.214188e+08  56115.718426   
282623        China   NaN          China  1.371220e+09   8027.683810   
282624        China   NaN          China  1.371220e+09   8027.683810   
282625        Spain   NaN          Spain  4.641827e+07  25831.582305   
282626      Germany   NaN        Germany  8.141314e+07  41313.313995   

        Score_medaille  
0                    0  
1                    0  
2                    0  
3                    3  
4                    0  
...                ...  
282622               3  
282623               1  
282624               3  
282625               2  
282626               3  

[282627 rows x 20 columns]

In [80] used -26.6 MiB RAM in 0.15s (system mean cpu 15%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 40.3 MiB


In [26]:
#Marche aussi 
''' 
medaille_categories=['Pas_medaille','Bronze','Silver','Gold']
encode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])

df_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])

df_complete
''' 

" \nmedaille_categories=['Pas_medaille','Bronze','Silver','Gold']\nencode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])\n\ndf_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])\n\ndf_complete\n"

In [26] used -3.6 MiB RAM in 0.11s (system mean cpu 17%, single max cpu 100%), peaked 3.6 MiB above final usage, current RAM usage now 333.0 MiB


## Analyse valeurs nulles 

In [81]:
df_complete.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
Country_x            419
notes             277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
dtype: int64

In [81] used 51.4 MiB RAM in 0.26s (system mean cpu 22%, single max cpu 46%), peaked 0.0 MiB above final usage, current RAM usage now 91.7 MiB


156 sports où on sait pas ce que c'est .... et des country (26 221) qui ne sont pas attribués non plus 

Pour les Country , on va merge 

In [82]:
df_complete_test=df_complete.copy()

df_complete_test=pd.merge(df_complete_test,df_regions,on='NOC',how='left')

df_complete_test['Country']=df_complete_test['Country_y'].fillna(df_complete_test['Country_x'])

In [82] used 126.0 MiB RAM in 0.33s (system mean cpu 15%, single max cpu 64%), peaked 0.0 MiB above final usage, current RAM usage now 217.7 MiB


In [84]:
df_complete_test.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
Country_x            419
notes_x           277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
Country              419
notes_y           277510
dtype: int64

In [84] used -136.0 MiB RAM in 0.29s (system mean cpu 15%, single max cpu 46%), peaked 0.0 MiB above final usage, current RAM usage now 133.6 MiB


In [56]:
liste_pays=df_regions['region'].unique().tolist()
liste_NOC=df_regions['NOC'].unique().tolist()

dictio_NOC_pays=dict(zip(liste_NOC,liste_pays))

print(dictio_NOC_pays)

df_complete_test['Country']=df_complete_test['Country'].fillna(df_complete['NOC'].map(dictio_NOC_pays))
#liste_region

df_complete_test

{'AFG': 'Afghanistan', 'AHO': 'Curacao', 'ALB': 'Albania', 'ALG': 'Algeria', 'AND': 'Andorra', 'ANG': 'Angola', 'ANT': 'Antigua', 'ANZ': 'Australia', 'ARG': 'Argentina', 'ARM': 'Armenia', 'ARU': 'Aruba', 'ASA': 'American Samoa', 'AUS': 'Austria', 'AUT': 'Azerbaijan', 'AZE': 'Bahamas', 'BAH': 'Bangladesh', 'BAN': 'Barbados', 'BAR': 'Burundi', 'BDI': 'Belgium', 'BEL': 'Benin', 'BEN': 'Bermuda', 'BER': 'Bhutan', 'BHU': 'Bosnia and Herzegovina', 'BIH': 'Belize', 'BIZ': 'Belarus', 'BLR': 'Czech Republic', 'BOH': 'Boliva', 'BOL': 'Botswana', 'BOT': 'Brazil', 'BRA': 'Bahrain', 'BRN': 'Brunei', 'BRU': 'Bulgaria', 'BUL': 'Burkina Faso', 'BUR': 'Central African Republic', 'CAF': 'Cambodia', 'CAM': 'Canada', 'CAN': 'Cayman Islands', 'CAY': 'Republic of Congo', 'CGO': 'Chad', 'CHA': 'Chile', 'CHI': 'China', 'CHN': 'Ivory Coast', 'CIV': 'Cameroon', 'CMR': 'Democratic Republic of the Congo', 'COD': 'Cook Islands', 'COK': 'Colombia', 'COL': 'Comoros', 'COM': 'Cape Verde', 'CPV': 'Costa Rica', 'CRC': 

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season  ...  \
0                            China  CHN  1992 Summer  1992  Summer  ...   
1                            China  CHN  2012 Summer  2012  Summer  ...   
2                          Denmark  DEN  1920 Summer  1920  Summer  ...   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer  ...   
4                      Netherlands  NED  1988 Winter  1988  Winter  ...   
...                            ...  ...          ...   ...     ...  ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer  ...   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer  ...   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer  ...   
282625                       Spain  ESP   Tokyo 2020  2020  Summer  ...   
282626                     Germany  GER   Tokyo 2020  2020  Summer  ...   

                      Sport                             Event         Medal  \
0                Basketball       Basketball Men's Basketball  Pas_medaille   
1                      Judo      Judo Men's Extra-Lightweight  Pas_medaille   
2                  Football           Football Men's Football  Pas_medaille   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War          Gold   
4             Speed Skating  Speed Skating Women's 500 metres  Pas_medaille   
...                     ...                               ...           ...   
282622            Taekwondo                       Women -57kg          Gold   
282623  Artistic Gymnastics                        Men's Team        Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars          Gold   
282625             Football                               Men        Silver   
282626               Tennis                     Men's Singles          Gold   

       notes_x      Country    Population  PIB/habitant  Score_medaille  \
0          NaN  Ivory Coast  1.371220e+09   8027.683810               0   
1          NaN  Ivory Coast  1.371220e+09   8027.683810               0   
2          NaN     Dominica  5.676002e+06  51989.293471               0   
3          NaN     Dominica  5.676002e+06  51989.293471               3   
4          NaN     Pakistan  1.693652e+07  44299.768085               0   
...        ...          ...           ...           ...             ...   
282622     NaN          NaN  3.214188e+08  56115.718426               3   
282623     NaN  Ivory Coast  1.371220e+09   8027.683810               1   
282624     NaN  Ivory Coast  1.371220e+09   8027.683810               3   
282625     NaN     Ethiopia  4.641827e+07  25831.582305               2   
282626     NaN       Guinea  8.141314e+07  41313.313995               3   

             region notes_y  
0             China     NaN  
1             China     NaN  
2           Denmark     NaN  
3           Denmark     NaN  
4       Netherlands     NaN  
...             ...     ...  
282622          USA     NaN  
282623        China     NaN  
282624        China     NaN  
282625        Spain     NaN  
282626      Germany     NaN  

[282627 rows x 21 columns]

In [56] used -14.1 MiB RAM in 0.40s (system mean cpu 22%, single max cpu 64%), peaked 0.0 MiB above final usage, current RAM usage now 188.0 MiB


In [44]:
df_complete_test.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
notes_x           277510
Country              419
Population         27593
PIB/habitant       35919
Score_medaille         0
region               419
notes_y           277510
dtype: int64

In [44] used 63.3 MiB RAM in 0.28s (system mean cpu 19%, single max cpu 54%), peaked 0.0 MiB above final usage, current RAM usage now 83.4 MiB


In [47]:
df_complete_test[df_complete_test['Country'].isna()]

Name Sex   Age  Height  Weight  \
578                    Zarinah Abdullah   F  21.0     NaN     NaN   
579                    Zarinah Abdullah   F  25.0     NaN     NaN   
2719    Saiyidah Aisyah Mohamed Rafa'ee   F  28.0   173.0    68.0   
6267                Paulo Amotun Lokoro   M  24.0   170.0    61.0   
7659                     Ang Peng Siong   M  21.0   180.0    75.0   
...                                 ...  ..   ...     ...     ...   
279101                       TAN Sze En   F  21.0     1.7     NaN   
279152             TCHATCHET II Cyrille   M  26.0     NaN     NaN   
279797             WAIS Ahmad Badreddin   M  30.0     NaN     NaN   
280196                      YEO Jia Min   F  22.0     NaN     NaN   
280240                        YU Mengyu   F  32.0     NaN     NaN   

                            Team  NOC        Games  Year  Season  ...  \
578                    Singapore  SGP  1992 Summer  1992  Summer  ...   
579                    Singapore  SGP  1996 Summer  1996  Summer  ...   
2719                   Singapore  SGP  2016 Summer  2016  Summer  ...   
6267    Refugee Olympic Athletes  ROT  2016 Summer  2016  Summer  ...   
7659                   Singapore  SGP  1984 Summer  1984  Summer  ...   
...                          ...  ...          ...   ...     ...  ...   
279101                 Singapore  SGP   Tokyo 2020  2020  Summer  ...   
279152                       EOR  EOR   Tokyo 2020  2020  Summer  ...   
279797                       EOR  EOR   Tokyo 2020  2020  Summer  ...   
280196                 Singapore  SGP   Tokyo 2020  2020  Summer  ...   
280240                 Singapore  SGP   Tokyo 2020  2020  Summer  ...   

                      Sport                                Event  \
578               Badminton            Badminton Women's Singles   
579               Badminton            Badminton Women's Singles   
2719                 Rowing         Rowing Women's Single Sculls   
6267              Athletics         Athletics Men's 1,500 metres   
7659               Swimming  Swimming Men's 100 metres Freestyle   
...                     ...                                  ...   
279101  Artistic Gymnastics                                  NaN   
279152        Weightlifting                                  NaN   
279797         Cycling Road                                  NaN   
280196            Badminton                                  NaN   
280240         Table Tennis                                  NaN   

               Medal notes_x Country Population  PIB/habitant  Score_medaille  \
578     Pas_medaille     NaN     NaN        NaN           NaN               0   
579     Pas_medaille     NaN     NaN        NaN           NaN               0   
2719    Pas_medaille     NaN     NaN        NaN           NaN               0   
6267    Pas_medaille     NaN     NaN        NaN           NaN               0   
7659    Pas_medaille     NaN     NaN        NaN           NaN               0   
...              ...     ...     ...        ...           ...             ...   
279101  Pas_medaille     NaN     NaN        NaN           NaN               0   
279152  Pas_medaille     NaN     NaN        NaN           NaN               0   
279797  Pas_medaille     NaN     NaN        NaN           NaN               0   
280196  Pas_medaille     NaN     NaN        NaN           NaN               0   
280240  Pas_medaille     NaN     NaN        NaN           NaN               0   

        region notes_y  
578        NaN     NaN  
579        NaN     NaN  
2719       NaN     NaN  
6267       NaN     NaN  
7659       NaN     NaN  
...        ...     ...  
279101     NaN     NaN  
279152     NaN     NaN  
279797     NaN     NaN  
280196     NaN     NaN  
280240     NaN     NaN  

[419 rows x 21 columns]

In [47] used 14.9 MiB RAM in 0.18s (system mean cpu 14%, single max cpu 22%), peaked 0.0 MiB above final usage, current RAM usage now 41.9 MiB


In [30]:
liste_sports=df_complete['Sport'].unique().tolist()

print(liste_sports)

['Basketball', 'Judo', 'Football', 'Tug-Of-War', 'Speed Skating', 'Cross Country Skiing', 'Athletics', 'Ice Hockey', 'Swimming', 'Badminton', 'Sailing', 'Biathlon', 'Gymnastics', 'Art Competitions', 'Alpine Skiing', 'Handball', 'Weightlifting', 'Wrestling', 'Luge', 'Water Polo', 'Hockey', 'Rowing', 'Bobsleigh', 'Fencing', 'Equestrianism', 'Shooting', 'Boxing', 'Taekwondo', 'Cycling', 'Diving', 'Canoeing', 'Tennis', 'Modern Pentathlon', 'Figure Skating', 'Golf', 'Softball', 'Archery', 'Volleyball', 'Synchronized Swimming', 'Table Tennis', 'Nordic Combined', 'Baseball', 'Rhythmic Gymnastics', 'Freestyle Skiing', 'Rugby Sevens', 'Trampolining', 'Beach Volleyball', 'Triathlon', 'Ski Jumping', 'Curling', 'Snowboarding', 'Rugby', 'Short Track Speed Skating', 'Skeleton', 'Lacrosse', 'Polo', 'Cricket', 'Racquets', 'Motorboating', 'Military Ski Patrol', 'Croquet', 'Jeu De Paume', 'Roque', 'Alpinism', 'Basque Pelota', 'Aeronautics', 'Cycling Road', 'Artistic Gymnastics', 'Karate', 'Trampoline Gy

In [31]:
table_sport=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sport

Sport  Name
0   3x3 Basketball    56
1      Aeronautics     1
2    Alpine Skiing  8829
3         Alpinism    25
4          Archery  2461
..             ...   ...
79      Tug-Of-War   170
80      Volleyball  3666
81      Water Polo  4119
82   Weightlifting  4129
83       Wrestling  7426

[84 rows x 2 columns]

In [31] used 2.4 MiB RAM in 0.16s (system mean cpu 28%, single max cpu 83%), peaked 0.0 MiB above final usage, current RAM usage now 337.4 MiB
